pkg/mutation/mutator/mutator.go

```diff
@@ -0,0 +1,63 @@
+package mutator
```

Comment:
Should these types belong to the mutation system package?

As it's the main consumer of these interfaces, it should at least control those definitions.

Reply:
Yes I was a bit torn about this.
Will move them.

Reply:
@maxsmythe as a result of moving it to the mutation package, I had to move the schema package to to avoid circular recursion.

Reply:
Ah, unfortunate. IMO we can work with things as-is and refactor them later.